### PR TITLE
Retry SubmitSavedClaimJob

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Summary

- Adds retries to the sidekiq job `SubmitSavedClaimJob`
- Adds additional logging to `SubmitSavedClaimJob`

## Related issue(s)
[department-of-veterans-affairs/va.gov-team#69027](https://github.com/department-of-veterans-affairs/va.gov-team/issues/69027)

## Testing done

- ran `tail -f log/development.log` in the terminal while submitting to [Pensions](http://localhost:3001/pension/application/527EZ/introduction). 
- When the request threw an error, the message appeared in the logs with a note that it was retrying; it retried several times.
- set the retry count to 0, and when the request failed, the error message appeared in the logs that all attempts had failed. 
- When the job succeeded, the success message appeared in the logs
      - *note*: I couldn't get the job to succeed locally without removing the external request (probably a mocking issue), but was able to test this in the review build by SSHing in and running `cd ~/vets-api; tail -f log/development.log`

## Screenshots
Info log on attempted submission:
Screenshot 2023-11-08 at 11 30 00 AM

Info log in response to error, and beginning retry:
Screenshot 2023-11-08 at 10 32 31 AM

Error log after all retries failed:
Screenshot 2023-11-08 at 10 57 11 AM

Info log on success:
Screenshot 2023-11-08 at 11 29 44 AM

## What areas of the site does it impact?
Multiple forms are handled by the SubmitSavedClaimJob, including Pension, Burial, EducationCareerCounseling, and VeteranReadinessEmployment. 

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Error in logs showing distinct logs for retrying vs exhausted Pension/Burial submission

### CODE REVIEW

**Comment by rmtolmach** [comment]
@TaiWilkin To get the **Validate changes to DataDog Service Catalog Files** check passing, please update your branch with the latest changes from `master`. Thanks!

**Comment by micahaspyr** [comment]
@TaiWilkin this looks great!  Was OCTO ok with the retries?  I feel like 14 retries at 25 is sufficient but just making sure they're cool with it.

**Comment by mjknight50** [comment]
> @TaiWilkin this looks great! Was OCTO ok with the retries? I feel like 14 retries at 25 is sufficient but just making sure they're cool with it.

I gave Tai that number based upon this conversation, for better or worse:
https://dsva.slack.com/archives/C053U7BUT27/p1697743066917679

**Comment by micahaspyr** [comment]
> > @TaiWilkin this looks great! Was OCTO ok with the retries? I feel like 14 retries at 25 is sufficient but just making sure they're cool with it.
> 
> I gave Tai that number based upon this conversation, for better or worse: https://dsva.slack.com/archives/C053U7BUT27/p1697743066917679

I think that timeframe should be ok unless we see some crazy outage that lasts more than a day

**Review by micahaspyr — CHANGES_REQUESTED** [review]

**Review by micahaspyr — APPROVED** [review]
This LGTM.  Good job!

File: app/sidekiq/central_mail/submit_saved_claim_job.rb
Comment (danlim715) [inline app/sidekiq/central_mail/submit_saved_claim_job.rb:62]: Would using ||= be useful here, and for all the instance variable instantiations? I'm not sure if this info persists between retry attempts, but if it does persist between retries, this could help reduce redundant DB calls, running of logic, etc. if it was successful the first time. If not, then please ignore my suggestion.

**Review by danlim715 — APPROVED** [review]
Added a quick question, but otherwise looks good to me. New specs also look like it has good coverage with new logic and that spec file fully passes locally for me.

**Review by wayne-weibel — APPROVED** [review]

File: app/sidekiq/central_mail/submit_saved_claim_job.rb
Comment (TaiWilkin) [reply app/sidekiq/central_mail/submit_saved_claim_job.rb:62]: I took a look into this, and it looks like the instance variables don't persist between retry attempts.

**Comment by TaiWilkin** [comment]
@department-of-veterans-affairs/lighthouse-banana-peels Can I get a review from someone on your team, as code owners? Thank you!

**Review by TS000 — APPROVED** [review]
🦉 Looks great.

**Comment by TaiWilkin** [comment]
Thank you for reviewing!

**Pull request merged**
